docs/dev_cycle.part-04.rst

```diff
@@ -0,0 +1,315 @@
+Part 4: Transaction Tests
+=========================
+
+.. contents:: :local:
+
+Test a Contract Function
+------------------------
+
+Edit the tests file and add another test:
+
+.. code-block:: shell
+
+    $ nano contracts/test_donator.py
+
+After the edit, the file should look as follows:
+
+.. code-block:: python
+
+
+    def test_default_usd_rate(chain):
+        donator, deploy_tx_hash = chain.provider.get_or_deploy_contract('Donator')
+        default_usd_rate = donator.call().default_usd_rate()
+        assert default_usd_rate == 350
+
+
+    def test_donate(chain):
+        donator, deploy_tx_hash = chain.provider.get_or_deploy_contract('Donator')
+        donator.transact({'value':500}).donate(370)
+        donator.transact({'value':650}).donate(380)
```

Comment:
This should be modified to include saving the transaction hash and waiting for the transaction to be mined since this will only pass on a `tester` chain as it is.

Reply:
will add in another place later, to keep things simple at the stage